### PR TITLE
Feat/340 card graphics

### LINKED ISSUE #340: Ability to use icon (fontawesome or similar) as card graphics.
Cards are more visually appealing when they include a graphic, but it can be time consuming to source an appropriate graphic, review terms of use, etc. In many cases, it would be acceptable or preferable to use a generic icon in place of a specific graphic.

One idea is to include fontawesome and/or simliar icon sets in the DevHub and provide a "shortcode" for referencing them in an image property such as `'image': 'fa:bicycle'`.  The `fa` prefix/namespace would cause the DevHub build process to use the corresponding fontawesome icon on the corresponding card in the built site.

### PR DESCRIPTION
**Pull request by patricksimonian**
## Summary
Update Cards and Avatars to use the NameSpacedImg component

This allows users to name space their devhub content for images

ie 
```md
---
author: github:patricksimonian
image:  fontawesome:book
resourceType: Documentation
---
```
The `author` and `image` are parsed by the `NameSpacedImg` component and render out an icon 
`<React-Avatar />` component or an `<i>` tag respectively. 

I would hold off on merging this PR as there is quite the chore to adjust all the external content to be namespaced appropriately. 

Fixes #340 
/bot-ignore-length

### CODE REVIEW

File: app-web/src/components/ToolsMenu/ToolsMenu.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { navigate } from 'gatsby';
 import { SEARCH } from '../../constants/ui';
+import NameSpacedImg from '../UI/NameSpacedImg/NameSpacedImg';
```
Comment (ShellyXueHan) [inline app-web/src/components/ToolsMenu/ToolsMenu.js]: where is the usage?

File: app-web/src/components/UI/NameSpacedImg/NameSpacedImg.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    return getNameSpacedComponent(namespace, imgSrc, rest);
+  } else if (/\.svg/.test(src) || src.indexOf('data:image') === 0) {
+    // is it an svg?
+    return <img src={src} {...rest} alt={alt} />;
+  } else if (/^https?/.test(src) || /\.(png|jpg|gif|jpeg)/.test(src)) {
+    // is it just a regular image?
+    return <Avatar src={src} {...rest} alt={alt} />;
+  } else {
+    // is it just a name?
+    return <Avatar name={src} {...rest} alt={src} />;
```
Comment (ShellyXueHan) [inline app-web/src/components/UI/NameSpacedImg/NameSpacedImg.js]: in what case it's a name?

File: app-web/src/components/ToolsMenu/ToolsMenu.js (hunk 1 above)
Comment (patricksimonian) [reply app-web/src/components/ToolsMenu/ToolsMenu.js]: removed the line

File: app-web/src/components/UI/NameSpacedImg/NameSpacedImg.js (hunk 2 above)
Comment (patricksimonian) [reply app-web/src/components/UI/NameSpacedImg/NameSpacedImg.js]: added more comments to explain usage

**Comment by patricksimonian** [comment]
this was closed as there is a lot of work being done with card designs. We will rework this feature branch to allow for the 'ability' of namespacing images/authors but not implmenting it.

**Pull request closed without merging**